### ISSUE TITLE
[Android] Link's Awakening black flicker only with NCE enabled  

### ISSUE THREAD

**Issue by mnisius** [yuzu-emu/yuzu#12339]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Affected Build(s)

GitHub 157

### Description of Issue

The Legend of Zelda Link's Awakening has very ugly black flicker when NCE is enabeld. With JIT there is no flicker but the frame rate is bad so it's also not playable. 

The flicker is present with Turnip and also with the Qualcomm Driver. 

This videos shows the flicker: 
https://discord.com/channels/398318088170242053/1113238919304204359/1183933616745627709

### Expected Behavior

No flicker 

### Reproduction Steps

Enable NCE, start the game 

### Log File

[yuzu_log.txt.old.txt](https://github.com/yuzu-emu/yuzu/files/13642562/yuzu_log.txt.old.txt)

### System Configuration

Samsung S23 Ultra 
8gen2

**Comment by chrisballensky** [comment]
Have seen this since the addition of NCE on early access builds. It's gradually gotten better, as the game wouldn't load initially with the new updates; but now flickers constantly regardless of driver.

**Comment by jwalsh1998** [comment]
Issue also occur on my s23 ultra. Seems to be a widespread problem.

**Comment by darrenhaken** [comment]
Anyone managed to fix this?

**Comment by mnisius** [comment]
There is no fix. You can disable nce but then performance will be bad. I think a developer needs to have a look at this.

**Comment by RainZSunday** [comment]
As of the latest build (170), I can confirm this is still an issue.

The flickering is so bad the game is unplayable, I tried any possible combination for the graphic settings and I've tried different drivers but it made no difference.

The only solution is switching off NCE, however, the FPS in the JIT version is far lower.

S23 ultra.

**Comment by nandolv** [comment]
Same issue with build 188 and lastest drivers on S23 Ultra...

**Comment by evil-trainer** [comment]
> As of the latest build (170), I can confirm this is still an issue.
> 
> The flickering is so bad the game is unplayable, I tried any possible combination for the graphic settings and I've tried different drivers but it made no difference.
> 
> The only solution is switching off NCE, however, the FPS in the JIT version is far lower.
> 
> S23 ultra.

If you want better FPS on JIT, set your emulation speed to 200% or even 400%.

**Comment by evil-trainer** [comment]
> Same issue with build 188 and lastest drivers on S23 Ultra...

Yeah, this is still a thing. And this game is still lacking a lot of graphical elements, so it's still a big no at the moment.

**Comment by williamsandonz** [comment]
Anyone had any luck with this lately? I'm itching to play

**Comment by williamsandonz** [comment]
Thanks for the update. I'm really uninformed about how all this all works but are we sure that we can safely rule out Yuzu android as not being the source of the issue purely because the PC version works? Assuming it is different source code and some things may not yet be implemented. (

**Comment by evil-trainer** [comment]
> > > Anyone had any luck with this lately? I'm itching to play
> > 
> > 
> > Nope, nothing changed. Mind you, this is a driver issue not a Yuzu problem. The game runs perfectly fine on Yuzu PC, so I guess this issue here must be closed and we must file a bug report to MESA so they can fix Turnip on this matter.
> 
> No it's not a driver problem. The game renders fine if you disable nce. And it doesn't matter if you User qualcom or Mesa driver.

The game DOES NOT render correctly even on JIT. See it yourself. One of the devs already said in Discord that it's a driver problem. If the game runs fine on PC and not on Android, it's most likely a driver issue since the code for both versions of the emulator is the same.

![1](https://github.com/yuzu-emu/yuzu/assets/149848175/337b3dd2-8ab5-4951-95e1-b25ef9968b20)
![2](https://github.com/yuzu-emu/yuzu/assets/149848175/d4dfebb2-e924-4880-ba81-288fbd0a740b)

**Comment by mnisius** [comment]
Yes some effects are missing but that wouldn't stop me from playing the game. The black flicker is what's stopping me. 

But the black flicker disappears if you turn off nce. So I would suggest it's related to nce.

**Comment by evil-trainer** [comment]
> Yes some effects are missing but that wouldn't stop me from playing the game. The black flicker is what's stopping me.
> 
> But the black flicker disappears if you turn off nce. So I would suggest it's related to nce.

Yeah, that's an NCE issue. Actually, I just realized I mixed the issues here and started talking about the missing effects that ARE a driver issue. Sorry about that 😅😅😅

And about the NCE flickering: yes byte[], one of the devs said once that he kinda knows how to fix it but it would lead to many other issues that users could rant even more about. So I guess he is not prioritizing it for now.

**Comment by evil-trainer** [comment]
> Thanks for the update. I'm really uninformed about how all this all works but are we sure that we can safely rule out Yuzu android as not being the source of the issue purely because the PC version works? Assuming it is different source code and some things may not yet be implemented. (

The source code is not different when it comes to rendering the game but yes, there are specific codes to android and PC. Which is the case for NCE backend. If you're ok with playing a game with so many graphics effects missing that ruin the overall experience (well, at least for me), just turn NCE off and play the game on JIT.

**Comment by williamsandonz** [comment]
Nice one thanks for clarifying all! So summary is:

1. Flicker caused by NCE. Fixeable by Yuzu Android team but not currently prioritised.
2. Missing effects issue caused by driver.

For me personally the flicker is definitely the biggest issue, I can deal with some missing effects. I also have an older phone (Red Magic 6S Pro) so not sure how well it would perform with NCE disabled but maybe I'll give it a go). I would have assume the NCE flicker issue would be widespread on other games and therefore a priority but sounds like it is somewhat specific to Links awakening.

**Comment by evil-trainer** [comment]
> Nice one thanks for clarifying all! So summary is:
> 
> 1. Flicker caused by NCE. Fixeable by Yuzu Android team but not currently prioritised.
> 2. Missing effects issue caused by driver.
> 
> For me personally the flicker is definitely the biggest issue, I can deal with some missing effects. I also have an older phone (Red Magic 6S Pro) so not sure how well it would perform with NCE disabled but maybe I'll give it a go). I would have assume the NCE flicker issue would be widespread on other games and therefore a priority but sounds like it is somewhat specific to Links awakening.

Yeah, that specific NCE flickering seems to happen only with LZA (or maybe a handful of other games). If it's running slow on JIT, try running the game on 200% speed and see if it at least caps to 30fps. This way you may have a "good" experience with it.
